src/components/AccountDropdown/AccountDropdown.tsx

```diff
@@ -0,0 +1,105 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import {
+  Button,
+  ButtonProps,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuOptionGroup,
+  Stack,
+} from '@chakra-ui/react'
+import { useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+
+import { RawText } from '../Text'
+import { AccountChildOption } from './AccountChildOption'
+import { AccountSegment } from './AccountSegement'
+
+export type AccountItem = {
+  account: string
+  cryptoBalance: string
+  name: string
+  symbol: string
+}
+
+type AccountDropdownProps = {
+  onClick: (arg: string) => void
+  activeAccount: string | null
+  accounts: AccountItem[]
```

Comment:
i would strongly prefer if the external interface of this component just accepted `AccountID[]` here

from that, it's able to determine asset ids belong to accounts, assets, and market data from selectors internally

passing a pre-computed object here has two issues

* performance
* every consumer has to collate all of this information (market data, asset data) into this shape to pass to this component, when all it needs is an `AccountId`

Reply:
Yah totally, whichever works for y'all when you start hooking it up to data.